analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/DiagnosticVerifier.cs

```diff
@@ -261,6 +280,36 @@ private static void VerifyBuildErrors(ImmutableArray<Diagnostic> diagnostics, Co
 
         private static string IssueType(bool isPrimary) => isPrimary ? "primary" : "secondary";
 
+        internal class FileNameSourceText
+        {
+            public string FileName { get; set; }
+            public SourceText SourceText { get; set; }
+            public FileNameSourceText(string fileName, SourceText sourceText)
+            {
+                FileName = fileName;
+                SourceText = sourceText;
+            }
+
+            public FileNameIssueLocations ToIssueLocations() =>
+                new FileNameIssueLocations(FileName, IssueLocationCollector.GetExpectedIssueLocations(SourceText.Lines));
+
+            public static FileNameSourceText FromSyntaxTree(SyntaxTree syntaxTree) =>
+                new FileNameSourceText(syntaxTree.FilePath, syntaxTree.GetText());
+        }
+
+        internal class FileNameIssueLocations
+        {
+            public string FileName { get; set; }
+            public IList<IIssueLocation> IssueLocations { get; set; }
+            public FileNameIssueLocations(string fileName, IList<IIssueLocation> issueLocations)
+            {
+                FileName = fileName;
+                IssueLocations = issueLocations;
+            }
+
+            public static FileNameIssueLocations BuildErrorsFromSyntaxTree(SyntaxTree syntaxTree) =>
```

Comment:
Let's remove this method. It's used just once, I originally wanted to make it constructor, but that leads to naming issue => it's a sign that it shouldn't be here.

Reply:
I don't really like the result, but ok.